core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/JarResultBuildStep.java

```diff
@@ -603,19 +604,30 @@ private JarBuildItem buildThinJar(CurateOutcomeBuildItem curateOutcomeBuildItem,
         Path decompiledOutputDir = null;
         boolean wasDecompiledSuccessfully = true;
         Decompiler decompiler = null;
-        if (packageConfig.vineflower.enabled) {
-            decompiledOutputDir = buildDir.getParent().resolve("decompiled");
-            FileUtil.deleteDirectory(decompiledOutputDir);
-            Files.createDirectory(decompiledOutputDir);
-            if (packageConfig.vineflower.enabled) {
+        if (packageConfig.vineflower.enabled.isPresent() || packageConfig.decompiler.enabled.isPresent()) {
```

Comment:
I think that you can just remove this `if` stmt because you check the enablement below...

Reply:
Done